[Documentation] Learning README update - Go references

Olshansk opened the pull request:
## Description

Updating the main README for new hires (internal or external) with good references on how to write idiomatic Go.

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [x] GitHub configuration

## List of changes

Adding additional links for good Go references including:
- Effective Go
- Practical Go

Olshansk committed:
Update learning README

Add go learning references

Olshansk commented:
@Gustavobelfort Can you PTAL at this PR when you have a chance? Just a couple of lines long.

Gustavobelfort reviewed (approved):
@Olshansk Can we add the standard project structure as well to aid the navigation of the folders and what each one does ? That would help golang newcomers

https://github.com/golang-standards/project-layout

Other than that LGTM

Olshansk committed:
Update learning README (#375)

## Description

Added a few notes, threads and blog posts that can help new contributors onboard

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [X] Documentation
- [ ] Other <!-- add details here if it a different type of change -->

## List of changes

- Added notes to provide context for how to approach the documentation & codebase for new V1 contributors
- Added a link to a blog post on modular blockchains - good article to build foundation for why we are building an L1 from scratch
- Added a link to the Anoma whitepaper - good long-term thinking
- Added links to github discussions around Tendermint rollbacks - good context for what we need to build in V1 and why we're moving away from the Cosmos implementation of Tendermint

Olshansk commented:
> @Olshansk Can we add the standard project structure as well to aid the navigation of the folders and what each one does ? That would help golang newcomers
> 
> [golang-standards/project-layout](https://github.com/golang-standards/project-layout)
> 
> Other than that LGTM

Great idea @Gustavobelfort, I support this 💯.

I created #379 to track this effort. You think you could take it on?
